gui/wxpython/datacatalog/tree.py

```diff
@@ -889,38 +980,21 @@ def OnEndDrag(self, node, event):
             self.OnPasteMap(event)
 
     def OnSwitchLocationMapset(self, event):
+        """Switch to location and mapset"""
+        self._SwitchLocationMapset()
+
+    def _SwitchLocationMapset(self):
+        """Switch to location and mapset"""
         genv = gisenv()
         if self.selected_location[0].label == genv['LOCATION_NAME']:
             self.changeMapset.emit(mapset=self.selected_mapset[0].label)
         else:
-            self.changeLocation.emit(mapset=self.selected_mapset[0].label, location=self.selected_location[0].label)
+            self.changeLocation.emit(mapset=self.selected_mapset[0].label,
+                                     location=self.selected_location[0].label)
         self.UpdateCurrentLocationMapsetNode()
         self.ExpandCurrentMapset()
         self.RefreshItems()
```

Comment:
This looks like a good start. Now you can maybe make use of the option of calling `_SwitchLocationMapset()` with different parameters based on the the context. However, still:

> I would need you to document your intentions here in the code to understand what this is supposed to solve. Documentation and comments describing the intent are often useful even when writing the code as they force us to think about the code is supposed to do. For example, for a line `add_to_list(abc)` we should aim at answering questions such as: Why we are adding to the list? What is the intended outcome? What are the possible outcomes? What are the assumptions about `abc` or other things before `add_to_list()` is called?

Reply:
Vashek, ok, I have a function OnRenameMapset that calls function rename_mapset_interactively and returns new mapset name. I think this is implemented right. 
But in the case of "if (self.selected_mapset[0].label == gisenv()['MAPSET'])" I need to switch my old mapset to the new one because my current mapset has been actually changed by renaming. But not literally so I need to self.selected_mapset[0].label = newmapset and then call the self._SwitchLocationMapset() function which can be generally the same for both functions OnRenameMapset, OnRenameLocation.
Where is the mistake in my thinking?

Reply:
...and _SwitchLocationMapset() uses the currently selected location/mapset to switch to. That all makes sense and should work. So, we have to continue to explore the actual behavior:

When I test renaming the current mapset in GUI, it freezes, but first it shows an error message "Error in g.mapset" saying that the mapset "old_name" does not exist. So, it seems g.mapset is used for the switch and it requires the old mapset to exist.

When I create the same situation in the command line and use g.mapset, I get the same behavior. Also I confirmed that g.mapset is used in ChangeLocationMapset() in datacatalog/frame.py to switch the mapset.

The conclusion here is that mapset cannot be changed when the current does not exist, or in our case, we cannot switch into the one with the new name after we renamed it as long as we are using g.mapset. g.mapset checks correctness of the state of the database connection, i.e., the current mapset, but we are of course breaking that correctness by our rename, so g.mapset is not the right tool for the task.

Therefore, for this ticket, use the same check to prevent user from renaming the current location or mapset. This is something which could be added in the future, but it is a separate issue/additional feature.

The freeze with "Error in g.mapset" window is possibly a separate issue too, but something we should get back to (at least open an issue - please do that if you can reproduce it outside of your PR, e.g., by deleting a the mapset directory in a file manager right before you switch to it in the data catalog).